src/lib.rs

```diff
@@ -19,18 +19,34 @@ pub use cstr_core as cstr;
 
 pub mod error;
 
+pub const BOARD: &'static str = match {
+    match cstr::CStr::from_bytes_with_nul(riot_sys::RIOT_BOARD) {
+        Ok(s) => s,
+        _ => panic!("Board names are null-terminated C strings"),
+    }
+
+    // Preferred would be
+    //     .expect("Board names are null-terminated C strings")
+}
+.to_str()
+{
+    Ok(x) => x,
+    // Preferred would be
+    // .expect("Board names should be ASCII");
+    // but feature(const_result) seems not to cover that yet
+    _ => panic!("Board names should be ASCII"),
+};
+
 /// Name of the RIOT board that is being used
 ///
 /// Development:
 ///
 /// Once this can be const, it'll be deprecated in favor of a pub const &'static str. That'll also
 /// force the compiler to remove all the exceptions at build time (currently it does not, even with
 /// aggressive optimization).
+#[deprecated(note = "Access BOARD instead")]
 pub fn board() -> &'static str {
```

Comment:
Alternatively this can be `pub const fn board() ...` and `pub const BOARD: &str = board();`.

Reply:
Right -- making it a const fn would really do just the same, especially when going with a const. (There'd be some difference when using a static, but I'd hope that the compiler stores a string valued const only once anyway if it detects multiple use).

Either way, [the stabilization is nigh](https://github.com/rust-lang/rust/pull/107429), 10 weeks to go :-)

Reply:
(Sorry for double-posting if this eventually does get through; I've written this already once and GitHub seems to have swallowed it).

I've tried this out on 1.72, and observed that unless forced at the user site through a local const (or using the feature inline_const), the exception only occurs at run time. That's probably fine from an execution model PoV (i.e., not a bug in Rust), but undesirable in our situation (eventually I hope to avoid all UTF-8 wellformedness checks in built code, but even if we can't get that, at least I don't want the error messages in my binaries). So I'd like to stick with having defining it in the const, and making board() -- indeed now a const fn -- return that value.